msg/templates/urtps/microRTPS_transport.cpp

```diff
@@ -113,104 +113,101 @@ ssize_t Transport_node::read(uint8_t *topic_ID, char out_buffer[], size_t buffer
 
 	*topic_ID = 255;
 
-	ssize_t len = node_read((void *)(rx_buffer + rx_buff_pos), sizeof(rx_buffer) - rx_buff_pos);
+	ssize_t len = 0;
 
-	if (len <= 0) {
-		int errsv = errno;
+	while (0 < (len = node_read((void *)(rx_buffer + rx_buff_pos), sizeof(rx_buffer) - rx_buff_pos))) {
```

Comment:
Why was this changed?

Reply:
This made more sense to me since we are using non-blocking reads, and overall the logic is kept the same.

Reply:
There's already a loop around the read call, so it's not required, but generally fine to keep it.
In which case is it executed more than once though? There's a `return len;` right at the bottom.

Reply:
Actually after your comment and a second thought on it, this while doesn't make sense. I just reverted it.